presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -356,21 +364,24 @@ public PlanOptimizers(
                                         new PruneCountAggregationOverScalar(metadata.getFunctionAndTypeManager()),
                                         new PruneOrderByInAggregation(metadata.getFunctionAndTypeManager()),
                                         new RewriteSpatialPartitioningAggregation(metadata)))
-                                .build()),
+                                .build(),
+                        Optional.of("PushDownAndMergeRules")),
                 new IterativeOptimizer(
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(
                                 new ImplementBernoulliSampleAsFilter(),
-                                new ImplementOffset())),
+                                new ImplementOffset()),
+                        Optional.of("ImplementBernoulliSampleAndOffset")),
                 simplifyOptimizer,
                 new UnaliasSymbolReferences(metadata.getFunctionAndTypeManager()),
                 new IterativeOptimizer(
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
-                        ImmutableSet.of(new RemoveRedundantIdentityProjections())),
+                        ImmutableSet.of(new RemoveRedundantIdentityProjections()),
+                        Optional.of("RemoveRedundantIdentityProjections")),
```

Comment:
Maybe you can use the classname or something and do it in the optimizer itself rather than the driver? 

Applies to all rules below

Reply:
Yeah, for non iterative optimizer rules, I am using the class name as the rule name. For the iterative optimizer, if the provided name is empty, it will default to use the concatenation of rule names included in the iterative optimizer. I previously thought it may be too long for some optimizers which have many rules. But maybe fine here, let me update the code then.